cfme/infrastructure/virtual_machines.py

```diff
@@ -991,39 +994,40 @@ def compare(self, *objects, **kwargs):
             sections: Which sections to compare.
             attributes: `all`, `different` or `same`. Default: `all`.
             mode: `exists` or `details`. Default: `exists`."""
-        sections = kwargs.get("sections")
-        attributes = kwargs.get("attributes", "all").lower()
-        mode = kwargs.get("mode", "exists").lower()
-        assert len(objects) >= 2, "You must specify at least two objects"
+        sections = kwargs.get('sections')
+        attributes = kwargs.get('attributes', 'all').lower()
+        mode = kwargs.get('mode', 'exists').lower()
+        assert len(objects) >= 2, 'You must specify at least two objects'
         objects = map(lambda o: o.name if isinstance(o, (Vm, Template)) else o, objects)
-        self.navigate()
+        view = self.navigate()
         for obj in objects:
             if not isinstance(obj, list):
-                path = self.genealogy_tree.find_path_to(obj)
-            self.genealogy_tree.check_node(*path)
-        toolbar.select("Compare selected VMs")
+                path = view.tree.find_path_to(obj)
+            view.tree.check_node(*path)
+        view.toolbar.compare.click()
+        view.flash.assert_no_errors()
         # COMPARE PAGE
-        flash.assert_no_errors()
+        compare_view = self.obj.create_view('Compare')
         if sections is not None:
-            map(lambda path: self.section_comparison_tree.check_node(*path), sections)
-            sel.click(self.apply_button)
-            flash.assert_no_errors()
+            map(lambda path: compare_view.tree.check_node(*path), sections)
+            compare_view.apply.click()
+            compare_view.flash.assert_no_errors()
         # Set requested attributes sets
-        toolbar.select(self.attr_mapping[attributes])
+        getattr(compare_view.toolbar, self.attr_mapping[attributes]).click()
         # Set the requested mode
-        toolbar.select(self.mode_mapping[mode])
+        getattr(compare_view.toolbar, self.mode_mapping[mode]).click()
 
     @property
     def tree(self):
         """Returns contents of the tree with genealogy"""
-        self.navigate()
-        return self.genealogy_tree.read_contents()
+        view = self.navigate()
+        return view.tree.read_contents()
 
     @property
     def ancestors(self):
         """Returns list of ancestors of the represented object."""
-        self.navigate()
-        path = self.genealogy_tree.find_path_to(re.compile(r"^.*?\(Selected\)$"))
+        view = self.navigate()
+        path = view.tree.find_path_to(re.compile(r"^.*?\(Selected\)$"))
```

Comment:
REQUIRED: please double check this code. WT tree doesn't have methods like find_path_to